### PR TITLE
Data Pipeline Refactor: Move SemanticSegmentation to new DataModule API

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

c.f title.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/image/segmentation/transforms.py [hunk 1]
```diff
@@ -74,3 +76,44 @@ def predict_default_transforms(image_size: Tuple[int, int]) -> Dict[str, Callabl
         ),
         "collate": kornia_collate,
     }
+
+
+def remove_extra_dimensions(batch: Dict[str, Any]):
```
Comment (tchaton) [inline flash/image/segmentation/transforms.py:81]: @ethanwharris Not sure 100% why the list is generated. Seems to be coming from the kornia_collate.

Comment (ethanwharris) [reply flash/image/segmentation/transforms.py:81]: Hmm, maybe it's an issue with the `ApplyToKeys` maybe worth to check that the list is length 1 just to make sure we don't lose any data

File: requirements/docs.txt [hunk 2]
```diff
@@ -13,4 +13,4 @@ sphinx-paramlinks>=0.5.1
 sphinx-togglebutton>=0.2
 sphinx-copybutton>=0.3
 
--r ../_notebooks/.actions/requirements.txt
+#-r ../_notebooks/.actions/requirements.txt
```
Comment (ethanwharris) [inline requirements/docs.txt:16]: I think we need these:
```suggestion
-r ../_notebooks/.actions/requirements.txt
```

**Review by ethanwharris — COMMENTED** [review]
LGTM, some comments

**Review by ethanwharris — APPROVED** [review]

File: flash/image/segmentation/transforms.py (hunk 1 above)
Comment (tchaton) [reply flash/image/segmentation/transforms.py:81]: Yes, it is of size 1

**Pull request merged**